distributed/shuffle/_merge.py

```diff
@@ -164,12 +168,13 @@ def merge_unpack(
     from dask.dataframe.multi import merge_chunk
 
     ext = _get_worker_extension()
+    # If the partition is empty, it doesn't contain the hash column name
     left = ext.get_output_partition(
         shuffle_id_left, barrier_left, output_partition
-    ).drop(columns=_HASH_COLUMN_NAME)
+    ).drop(columns=_HASH_COLUMN_NAME, errors="ignore")
```

Comment:
This is inelegant, but so is adding the hash column to the meta. If anybody has strong preferences, please speak up.

Reply:
This seems fine to me. Thanks for adding the informative comment 👌